client/lobbies/lobby.tsx

```diff
@@ -413,9 +222,10 @@ export default class Lobby extends React.Component {
     onRemoveObserver: PropTypes.func,
     onMapPreview: PropTypes.func,
     onToggleFavoriteMap: PropTypes.func,
+    onStartGame: PropTypes.func,
   }
 
-  getTeamSlots(team, isObserver, isLobbyUms) {
+  getTeamSlots(team: typeof Team, isObserver: boolean, isLobbyUms: boolean) {
```

Comment:
I think the places you're using `typeof` here are wrong (in a similar way to how the `typeof [WhateverMessage]` stuff was in the prior version of this PR). You probably intend `ReturnType<typeof Team>` or something (but ideally we would migrate these records to be class-extends form as well so this stuff is unnecessary, don't expect you to do that in this PR though)